packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts

```diff
@@ -37,20 +49,35 @@ export function trackInteractionToNextPaint(viewLoadingType: ViewLoadingType, li
       }
     }
 
-    const newInpDuration = longestInteractions.estimateP98Duration()
-    if (newInpDuration) {
-      inpDuration = newInpDuration
+    const newInteraction = longestInteractions.estimateP98Interaction()
+    if (newInteraction) {
+      interactionToNextPaint = newInteraction.duration
+      if (
+        isExperimentalFeatureEnabled(ExperimentalFeature.WEB_VITALS_ATTRIBUTION) &&
+        newInteraction.target &&
+        isElementNode(newInteraction.target)
+      ) {
+        interactionToNextPaintTargetSelector = getSelectorFromElement(
+          newInteraction.target,
+          configuration.actionNameAttribute
+        )
+      }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏if the target of a first interaction is an element, then the target of a second interaction is not an element, the selector still points to the first interaction. Reset the selector maybe?
```suggestion
      if (
        isExperimentalFeatureEnabled(ExperimentalFeature.WEB_VITALS_ATTRIBUTION) &&
        newInteraction.target &&
        isElementNode(newInteraction.target)
      ) {
        interactionToNextPaintTargetSelector = getSelectorFromElement(
          newInteraction.target,
          configuration.actionNameAttribute
        )
      } else {
        interactionToNextPaintTargetSelector = undefined
      }
```